sirepo/auth/email.py

```diff
@@ -150,11 +150,11 @@ def create_token(self):
         @classmethod
         def delete_changed_email(cls, user):
             with auth_db.thread_lock:
-                cls._session.query(cls).filter(
+                cls._session().query(cls).filter(
```

Comment:
I don't think this is right. I think this should be:

```py
with auth_db.session() as s:
    s.query(class).filter(...).delete()
```

This then also does the commit implicitly. Any time a session() call exits, it either rolls back (finally) or commits (normal flow).

Implicit in the session is an auth_db.thread_lock, too.

Reply:
I'm confused, doing this would create/destroy the session for every call. We discussed doing that but you said that we should have one session per request.

Reply:
We discussed leaving the auth_db.session() in uri_router._dispatch (every request) and then manually calling it when needed outside of flask (pkcli, tests, supervisor)